app/src/main/java/com/strv/movies/ui/moviedetail/MovieDetailViewModel.kt

```diff
@@ -0,0 +1,45 @@
+package com.strv.movies.ui.moviedetail
+
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.strv.movies.data.OfflineMoviesProvider
+import com.strv.movies.ui.navigation.MoviesNavArguments
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+import kotlin.random.Random
+
+@HiltViewModel
+class MovieDetailViewModel @Inject constructor(
+    savedStateHandle: SavedStateHandle,
+) : ViewModel() {
+
+    private val movieId =
+        requireNotNull(savedStateHandle.get<Int>(MoviesNavArguments.MOVIE_ID_KEY)) {
+            "Movie id is missing..."
+        }
+
+    private val _viewState = MutableStateFlow(MovieDetailViewState(loading = true))
+    val viewState = _viewState.asStateFlow()
```

Comment:
I wouldn't mind skipping the backing property here as it can be little confusing and doesn't add that much value to the students. 
But feel free to keep it if you are planning to explain it 👌

Reply:
I will explain it.